datago/src/client.rs

```diff
@@ -0,0 +1,224 @@
+use crate::generator_http;
+use crate::image_processing::ARAwareTransform;
+use crate::image_processing::ImageTransformConfig;
+use crate::worker_http;
+use pyo3::prelude::*;
+
+use kanal::bounded;
+use serde::Deserialize;
+use std::sync::Arc;
+use std::thread;
+use threadpool::ThreadPool;
+
+#[derive(Deserialize)]
+struct DatagoClientConfig {
+    source_config: generator_http::SourceDBConfig,
+    image_config: Option<ImageTransformConfig>,
+    limit: usize,
+    rank: usize,
+    world_size: usize,
+    samples_buffer_size: usize,
+}
+
+#[pyclass]
+pub struct DatagoClient {
+    pub is_started: bool,
+    source_config: generator_http::SourceDBConfig,
+    limit: usize,
+
+    // Perf settings
+    num_threads: usize,
+    max_connections: usize,
+    rank: usize,
+    world_size: usize,
+
+    // Channels
+    pages_tx: kanal::Sender<serde_json::Value>,
+    pages_rx: kanal::Receiver<serde_json::Value>,
+    samples_meta_tx: kanal::Sender<serde_json::Value>,
+    samples_meta_rx: kanal::Receiver<serde_json::Value>,
+    samples_tx: kanal::Sender<worker_http::Sample>,
+    samples_rx: kanal::Receiver<worker_http::Sample>,
+    worker_done_count: Arc<std::sync::atomic::AtomicUsize>,
+
+    // Sample processing
+    image_transform: Option<ARAwareTransform>,
+    encode_images: bool,
+
+    // Threads
+    pinger: Option<thread::JoinHandle<()>>,
+    feeder: Option<thread::JoinHandle<()>>,
+    thread_pool: ThreadPool,
+}
+
+#[pymethods]
+impl DatagoClient {
+    #[new]
+    pub fn new(str_config: String) -> Self {
+        let config: DatagoClientConfig = serde_json::from_str(&str_config).unwrap();
+
+        let (pages_tx, pages_rx) = bounded(2);
+        let (samples_meta_tx, samples_meta_rx) = bounded(config.samples_buffer_size);
+        let (samples_tx, samples_rx) = bounded(config.samples_buffer_size);
+
+        let mut image_transform: Option<ARAwareTransform> = None;
+        let mut encode_images = false;
+
+        if config.image_config.is_some() {
+            let image_config = config.image_config.unwrap();
+            println!("Cropping and resizing images");
+            image_transform = Some(image_config.get_ar_aware_transform());
+            encode_images = image_config.pre_encode_images;
+        }
```

Comment:
```suggestion
        if config.image_config.is_some() {
            println!("Cropping and resizing images");
        }
        
        let mut image_transform = config.image_config.map(|image_config| image_config.get_ar_aware_transform());
        let mut encode_images = config.image_config.map_or(false, |image_config| image_config.pre_encode_images);
```

Or it could also be written like (the goal is to avoid the unwrap)

```rust
        let mut image_transform: Option<ARAwareTransform> = None;
        let mut encode_images = false;

        if let Some(image_config) = config.image_config {
            println!("Cropping and resizing images");
            image_transform = Some(image_config.get_ar_aware_transform());
            encode_images = image_config.pre_encode_images;
        }
```